WooCommerce/Classes/ViewRelated/Orders/Collect Payments/CollectOrderPaymentUseCase.swift

```diff
@@ -703,6 +719,44 @@ private extension CollectOrderPaymentUseCase {
         }
         rootViewController.present(receiptEmailViewController, animated: true)
     }
+
+    private func receiptStateForFailedPayment(error: Error,
```

Comment:
I think this kind of code could be moved to some sort of `EmailReceiptCoordinator` that would control the presentation and presentation logic and potentially could be used elsewhere.